tests/unit/test_broker.py

```diff
@@ -0,0 +1,94 @@
+from unittest.mock import Mock, create_autospec
+
+import pytest
+from bugzilla import Bugzilla
+from jira import JIRA
+from pydantic.error_wrappers import ValidationError
+
+import bugjira.broker as broker
+from bugjira.broker import (
+    Broker,
+    BrokerInitException,
+    BugzillaBroker,
+    JiraBroker,
+)
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup(monkeypatch):
+    monkeypatch.setattr(broker, "Bugzilla", create_autospec(Bugzilla))
+    monkeypatch.setattr(broker, "JIRA", create_autospec(JIRA))
+
+
+def test_broker_init_no_backend():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it without supplying a backend parameter
+    THEN the resulting instance's backend attribute should be None
+    """
+    broker = Broker(config={})
+    assert broker.backend is None
+
+
+def test_broker_init_with_backend():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it with a backend parameter supplied
+    THEN the resulting instance's backend attribute should be the one we input
+        to the constructor
+    """
+    backend = Mock()
+    broker = Broker(backend=backend)
+    assert broker.backend == backend
+
+
+def test_broker_init_with_no_parameters():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it with no parameters
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(BrokerInitException):
+        Broker()
+
+
+def test_broker_init_with_no_config_and_no_backend():
+    """
+    GIVEN the Broker class' constructor
+    WHEN we call it with None for both parameters
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(BrokerInitException):
+        Broker(config=None, backend=None)
+
+
+def test_bugzilla_broker_init_with_valid_config(good_config_dict):
+    bzb = BugzillaBroker(config=good_config_dict)
+    assert bzb.backend.getbug
```

Comment:
This asserting to true means the getbug method exists, right? Not sure i have used this syntax, so wanted to verify. Also, think the docstring was missed on this one.

Reply:
That's right; I'm just using the method's presence (duck typing) to show that it quacks like a bugzilla api backend.